python/kserve/kserve/triton/triton_model.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 The KServe Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union, Dict, AsyncIterator, Any
+
+import tritonserver
+from prometheus_client import REGISTRY
+
+from .metrics import TritonMetricsCollector
+from .utils import create_triton_infer_request, to_infer_response
+from ..logging import logger
+from ..model import (
+    Model,
+    InferRequest,
+    InferResponse,
+)
+from ..protocol.grpc.grpc_predict_v2_pb2 import ModelInferRequest
+
+
+class TritonModel(Model):
+
+    def __init__(
+        self,
+        model_name: str,
+        triton_options: tritonserver.Options = None,
+    ):
+        super().__init__(model_name)
+        self._server: tritonserver.Server = None
+        self._model: tritonserver.Model = None
```

Comment:
Do we support multiple models from the repository or only one?

Reply:
Will add support for multiple models and support for other custom runtimes in separate PR.

Reply:
Let's create an issue to track that

Reply:
Created #3945